test/debugger.js

```diff
@@ -0,0 +1,266 @@
+    } else {
+      callback(null, body.breakpoint);
+    }
+  });
+};
+
+/**
+ * Sets a new breakpoint.
+ * @param {Debuggee} debuggeeId - The ID of the debuggee in which the breakpoint
+ *     should be set.
+ * @param {Breakpoint} breakpoint - An object representing information about
+ *     the breakpoint to set.
+ * @param {!function(?Error,Breakpoint=)} callback - A function that will be
+ *     called with information about the breakpoint that was just set, or an
+ *     Error object if an error occurred in obtaining its information. Note
+ *     that the Breakpoint object here will differ from the input object in
+ *     that its id field will be set.
+ */
+Debugger.prototype.setBreakpoint = function(debuggeeId, breakpoint, callback) {
+  var query = {
+    clientVersion: this.clientVersion_
+  };
+  var options = {
+    uri: API + '/debuggees/' + encodeURIComponent(debuggeeId) +
+         '/breakpoints/set?' + qs.stringify(query),
+    method: 'POST',
+    json: true,
+    body: breakpoint
+  };
+  
+  this.request(options, function(err, body, response) {
+    if (err) {
+      callback(err);
+    } else if (!response) {
+      callback(err || new Error('unknown error - request response missing'));
+    } else if (response.statusCode !== 200) {
+      callback(new Error('unable to set breakpoint, status code ' +
+                         response.statusCode));
+    } else if (!body || !body.breakpoint) {
+      callback(new Error('invalid response body from server'));
+    } else {
+      callback(null, body.breakpoint);
+    }
+  });
+};
+
+/**
+ * Deletes a breakpoint.
+ * @param {Debuggee} debuggeeId - The ID of the debuggee to which the breakpoint
+ *     belongs.
+ * @param {Breakpoint} breakpointId - The ID of the breakpoint to delete.
+ * @param {!function(?Error)} callback - A function that will be
+ *     called with a Error object as a parameter if an error occurred in
+ *     deleting a breakpoint. If no error occurred, the first argument will be
+ *     set to null.
+ */
+Debugger.prototype.deleteBreakpoint = function(debuggeeId, breakpointId, callback) {
+  var query = {
+    clientVersion: this.clientVersion_
+  };
+  var options = {
+    uri: API + '/debuggees/' + encodeURIComponent(debuggeeId) +
+         '/breakpoints/' + encodeURIComponent(breakpointId) + '?' +
+         qs.stringify(query),
+    method: 'DELETE',
+    json: true
+  };
+
+  this.request(options, function(err, body, response) {
+    if (err) {
+      callback(err);
+    } else if (!response) {
+      callback(err || new Error('unknown error - request response missing'));
+    } else if (response.statusCode !== 200) {
+      callback(new Error('unable to delete breakpoint, status code ' +
+                         response.statusCode));
+    } else if (Object.keys(body).length > 0) {
+      callback(new Error('response body is non-empty'));
+    } else {
+      callback(null);
+    }
+  });
+};
+
+module.exports = Debugger;
```

Comment:
Okay, this class is really well done, so I don't see a reason to keep it as a utility file for the tests. Please move this to `src/`.